code/builders/builder-vite/src/codegen-importfn-script.ts

```diff
@@ -1,7 +1,9 @@
-import { relative } from 'node:path';
-
 import type { Options } from 'storybook/internal/types';
 
+import { genDynamicImport, genImport, genObjectFromRawEntries } from 'knitwork';
```

Comment:
I'm not a fan of using this package for multiple reasons:

1. With our recent focus on slimming down Storybook, we're trying to ensure that every KB counts, and to me it doesn't look like this improves the code in any way. The library covers a lot of edge cases/options which I don't think is relevant to us - but maybe I'm missing something?
2. I feel like the readability suffers here, because now to understand what the code does, you need to dig into what `knitwork` is and does. Before it was just simple string concatenations, which you could understand by understanding basic JS.

But again, maybe this adds compatibility for edge cases that users are experiencing which I don't know about?

Reply:
> The package size of the builder should be the same (or actually slightly goes down since its shorter to write the code using knitwork over manual string concats).

From a bundled perspective yes, but not when you take into account that the extra dependencies need to be downloaded and installed.

> I also don't quite understand why the builder size would matter much, since it is never deployed to any server.

I'm sorry I didn't make this clear in the original comment, but for us it's not just about the size of the final bundle of a built Storybook, it's about the whole Storybook experience, which includes how fast it installs, how many dependencies it adds to your project, how many MBs it adds to your `node_modules`.
By far the biggest complaint about Storybook today is that it's "bloated" and people don't want to add the bloat to their projects.
You can very rightfully interpret that complaint however you like, but it doesn't change the fact that our users are loud and clear about caring about this, and so do we.

This is why dependencies need to be justified.

You can read more about the ethos at https://e18e.dev/guide/cleanup.html and our perspective on it in #29038.

> I would argue this is very subjective. Personally, I find `genDynamicImport(path)` more readable

Subjective yes. IMO `genDynamicImport` is only more readable if you understand what it does, which most don't.

> and it is less prone to make a small mistake

I can 100% agree with that.

It doesn't make sense to me to replace ``async () => import('${path}')` with a dependency.

Reply:
If you really think that reducing 30kb in node_modules for developers (not production) is worth the danger of introducing bugs by manual string concatenation, may I ask you to migrate the code away from knitwork (here as well as in https://github.com/storybookjs/storybook/pull/28798)?